src/librbd/cache/pwl/LogEntry.h

```diff
@@ -173,17 +190,20 @@ class WriteLogEntry : public GenericWriteLogEntry {
   #ifdef WITH_RBD_RWL
   void init_pmem_buffer(std::vector<WriteBufferAllocation>::iterator allocation);
   #endif
+  virtual void init_cache_bl(bufferlist &src_bl, uint64_t off, uint64_t len) {}
+  /* Returns a ref to a bl containing bufferptrs to the entry pmem buffer */
+  virtual buffer::list &get_cache_bl();
+
   BlockExtent block_extent();
   unsigned int reader_count() const;
-  /* Returns a ref to a bl containing bufferptrs to the entry pmem buffer */
-  buffer::list &get_pmem_bl();
   /* Constructs a new bl containing copies of pmem_bp */
   void copy_pmem_bl(bufferlist *out_bl) override;
```

Comment:
Nit: is there a (generic) way to eliminate references to pmem from the base class?

Reply:
Not sure I understand this - references to the word pmem? or the bufferlist itself? If it's the bufferlist - it's currently fetched in base classes `ReplicatedWriteLog` and `SSDWriteLog` to write the data to memory and disk respectively.

Reply:
`pmem` is tied to RWL so a generic base class shouldn't have references to `pmem`

Reply:
Done